File: lib/java/opentoken/src/main/java/com/truveta/opentoken/tokentransformer/TokenTransformer.java
/**
 * Copyright (c) Truveta. All rights reserved.
 */
package com.truveta.opentoken.tokentransformer;

import java.io.Serializable;

/**
 * A generic interface for the token transformer.
 */
public interface TokenTransformer extends Serializable {
    /**
     * Transforms the token using a token transformation rule/strategy.
     * 
     * @param token the token to be transformed.
     * 
     * @return the transformed token.
     * 
     * @throws Exception error encountered while transforming the token.
     */
    String transform(String token) throws Exception;
}

File: lib/python/opentoken/src/main/opentoken/tokentransformer/token_transformer.py
"""
Copyright (c) Truveta. All rights reserved.
"""

from abc import ABC, abstractmethod


class TokenTransformer(ABC):
    """A generic interface for the token transformer."""

    @abstractmethod
    def transform(self, token: str) -> str:
        """
        Transforms the token using a token transformation rule/strategy.

        Args:
            token: The token to be transformed.

        Returns:
            The transformed token.

        Raises:
            Exception: Error encountered while transforming the token.
        """
        pass
